src/test/java/com/github/tomakehurst/wiremock/SnapshotAcceptanceTest.java

```diff
@@ -0,0 +1,274 @@
+import static com.github.tomakehurst.wiremock.testsupport.TestHttpHeader.withHeader;
+import static com.github.tomakehurst.wiremock.testsupport.WireMatchers.equalToJson;
+import static junit.framework.TestCase.assertFalse;
+import static org.junit.Assert.assertThat;
+
+public class SnapshotAcceptanceTest extends AcceptanceTestBase {
+    private WireMockServer proxyingService;
+    private WireMockTestClient proxyingTestClient;
+
+    public void init() {
+        proxyingService = new WireMockServer(wireMockConfig()
+            .dynamicPort()
+            .withRootDirectory("src/test/resources/empty"));
+        proxyingService.start();
+        proxyingService.stubFor(proxyAllTo("http://localhost:" + wireMockServer.port()));
+
+        proxyingTestClient = new WireMockTestClient(proxyingService.port());
+        wireMockServer.stubFor(any(anyUrl()).willReturn(ok()));
+    }
+
+    @After
+    public void proxyServerShutdown() {
+        proxyingService.stop();
+    }
+
+    private static final String DEFAULT_SNAPSHOT_RESPONSE =
+            "[                                                           \n" +
+            "    {                                                       \n" +
+            "        \"request\" : {                                     \n" +
+            "            \"url\" : \"/foo/bar/baz\",                     \n" +
+            "            \"method\" : \"GET\"                            \n" +
+            "        },                                                  \n" +
+            "        \"response\" : {                                    \n" +
+            "            \"status\" : 200                                \n" +
+            "        }                                                   \n" +
+            "    },                                                      \n" +
+            "    {                                                       \n" +
+            "        \"request\" : {                                     \n" +
+            "            \"url\" : \"/foo/bar\",                         \n" +
+            "            \"method\" : \"GET\"                            \n" +
+            "        },                                                  \n" +
+            "        \"response\" : {                                    \n" +
+            "            \"status\" : 200                                \n" +
+            "        }                                                   \n" +
+            "    }                                                       \n" +
+            " ]                                                            ";
+
+    @Test
+    public void returnsRequestsWithDefaultOptions() throws Exception {
+        proxyingTestClient.get("/foo/bar", withHeader("A", "B"));
+        proxyingTestClient.get("/foo/bar/baz", withHeader("A", "B"));
+
+        assertThat(
+            proxyingTestClient.snapshot("{ \"persist\": false }"),
```

Comment:
That should be fixable just by adding this to the After method:

```java
 proxyingService.resetMappings();
```

That will delete persistent mappings (not obvious from the name, I know!).